inst/include/common/model.hpp

```diff
@@ -86,99 +84,171 @@ class Model {  // may need singleton
+        jnll += nll_components[nll_components_idx];
+        n_priors += 1;
+        nll_components_idx += 1;
+      }
+    }
+    MODEL_LOG << "The joint negative log likelihood after evaluating "
+              << n_priors << " prior distributions is:  "<< jnll << "."
+              << std::endl;
+
 
-    // Loop over populations, evaluate, and sum up the recruitment likelihood
-    // component
+    // Loop over populations and evaluate recruitment component
 
-    typename fims_info::Information<Type>::population_iterator it;
-    MODEL_LOG << "Evaluating expected values and summing recruitment nlls for "
+    typename fims_info::Information<Type>::population_iterator p_it;
+    MODEL_LOG << "Evaluating recruitment expected values for "
               << this->fims_information->populations.size() << " populations."
               << std::endl;
 
-    for (it = this->fims_information->populations.begin();
-         it != this->fims_information->populations.end(); ++it) {
-      //(*it).second points to the Population module
-      MODEL_LOG << "Setting up pointer to population " << (*it).second->id
+    for (p_it = this->fims_information->populations.begin();
+         p_it != this->fims_information->populations.end(); ++p_it) {
+      //(*p_it).second points to the Population module
+      std::shared_ptr<fims_popdy::Population<Type> > p = (*p_it).second;
+      MODEL_LOG << "Setting up pointer to population " << p->id
                 << "." << std::endl;
       // Prepare recruitment
-      (*it).second->recruitment->Prepare();
+      p->recruitment->Prepare();
       MODEL_LOG << "Recruitment for population successfully prepared"
                 << std::endl;
-// link to TMB objective function
-#ifdef TMB_MODEL
-      (*it).second->of = this->of;
-#endif
-      // Evaluate population
-      (*it).second->Evaluate();
-      // Recrtuiment negative log-likelihood
-      rec_nll -= (*it).second->recruitment->evaluate_lpdf();
-      MODEL_LOG << "Recruitment negative log-likelihood is: " << rec_nll
-                << std::endl;
     }
-    MODEL_LOG << "All populations successfully evaluated." << std::endl;
 
-    // Loop over fleets/surveys, and sum up age comp and index nlls
+    // Loop over densities and evaluate joint negative log-likelihoods for random effects
+    MODEL_LOG << "Setup random effects." << std::endl;
+    this->fims_information->setup_random_effects();
+    size_t n_random_effects = 0;
+    for(d_it = this->fims_information->density_components.begin();
+        d_it!= this->fims_information->density_components.end(); ++d_it){
+      std::shared_ptr<fims_distributions::DensityComponentBase<Type> > d = (*d_it).second;
+      #ifdef TMB_MODEL
+        d->of = this->of;
+      #endif
+      if(d->input_type == "random_effects"){
+        MODEL_LOG << "d->x.size is: " << d->x.size() << std::endl;
+        MODEL_LOG << "d->xexpected_values.size is: " << d->expected_values.size() << std::endl;
```

Comment:
There is a typo on line 145: remove the "x" in front of "expected_values.size is"

Reply:
Removed MODEL_LOG comments as they were used for debugging.